tests/test_precedence.rs

```diff
@@ -95,9 +95,9 @@ fn test_rustc_precedence() {
         let mut content = String::new();
         file.read_to_string(&mut content).unwrap();
 
-        let (l_passed, l_failed) = match content.parse::<syn::Crate>() {
-            Ok(krate) => {
-                let exprs = collect_exprs(krate);
+        let (l_passed, l_failed) = match syn::parse_str::<syn::File>(&content) {
```

Comment:
You'll need to update this to use the BOM and shebang-handling `syn::parse_file` function instead. Some of these input strings contain those characters.